### ISSUE TITLE
Tag for variable type behave inconsistent

### ISSUE THREAD

**Issue by pshemass** [zio/izumi-reflect#192]
I tried to use local variable type to distinct two variables with the same type in `ZIO` environment but it seems that even for the same variable `Tag` is different when we access variable in different way. 

```
object Foo {
  val bar = "bar"

  val t1 = Tag[bar.type]
  val t2 = Tag[Foo.bar.type]
  val t3 = Tag[Foo.this.bar.type]

  t1 == t2 // false
  t2 == t3 // false
  t3 == t1 // false
}
```

is this expected?

**Comment by pshirshov** [comment]
Well, in my case the following test passes:

```scala
      object Foo {
        val bar = "bar"

        val t1 = Tag[bar.type]
        val t2 = Tag[Foo.bar.type]
        val t3 = Tag[Foo.this.bar.type]
        println(t1.tag.repr)
        println(t2.tag.repr)
        println(t3.tag.repr)
      }

      import Foo._
      assert(t1.tag =:= t3.tag)
      assert(!(t2.tag =:= t3.tag))
```

it prints 

```
java.lang::izumi.reflect.test.TagTest.Foo.bar
izumi.reflect.test.TagTest.Foo::izumi.reflect.test.TagTest.Foo.bar
java.lang::izumi.reflect.test.TagTest.Foo.bar
```

This looks like a bug or implementation limitation. `java.lang` is clearly a wrong prefix for `bar`. And obviously it comes from String. I'm not sure yet if it's fixable and how easy it is to fix.

I've added a progression test for now.

**Comment by pshirshov** [comment]
Surprisingly, I've found a fix:

```diff
-            case _ =>
-              fromRef(tpef.typeSymbol.owner.asType.toType)
+            case ksymtype =>
+              fromRef(ksymtype)
```

in `def unpackPrefix(pre: Type): Option[AppliedReference]`

**Comment by pshirshov** [comment]
Actually this is a very strange expression: `tpef.typeSymbol.owner.asType.toType`. I have no idea why did I write this, it looks like it makes little sense in this context O_o

**Comment by pshirshov** [comment]
Well, actually it wasn't me. That was a workaround for some behaviour specified in `"consider class member's this-prefix to be the defining template, not the most specific prefix from where the call is happening (deliberate omission of this for better ergonomics in cakes)"`. That behaviour somehow contradicts to PDT machinery of Scala but makes tags more convenient. We'll think how to fix both these cases.

**Comment by pshirshov** [comment]
Well, now both the cases are fixed. I've found a couple more cornercases and added some progression tests though this particular issue is fixed.

**Comment by pshemass** [comment]
@pshirshov when do you think this will be released ?

**Comment by neko-kai** [comment]
@pshemass @pshirshov Released under version 2.0.0 https://github.com/zio/izumi-reflect/releases/tag/v2.0.0
